Json processdata fix for objects with zero-indexed keys (#18762)
yiisoft/yii2: framework/CHANGELOG.md

- Bug #18648: Fix `yii\web\Request` to properly handle HTTP Basic Auth headers (olegbaturin)
- Enh #18726: Added `yii\helpers\Json::$prettyPrint` (rhertogh)
- Enh #18734: Added `yii\validators\EmailValidator::$enableLocalIDN` (brandonkelly)

> - Enh #18762: Avoid changing zero-indexed objects to array in `yii\helpers\Json::encode()`. Add `yii\helpers\Json::$keepObjectType` and `keepObjectType` option for `yii\web\JsonResponseFormatter`. Disabled by default. (zebraf1)

Reviewer: ```suggestion
- Enh #18762: Added `yii\helpers\Json::$keepObjectType` and `yii\web\JsonResponseFormatter::$keepObjectType` in order to avoid changing zero-indexed objects to array in `yii\helpers\Json::encode()` (zebraf1)
```

Author: Changed
